src/common.cpp

```diff
@@ -1873,6 +1884,63 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    static std::string result;
+    if (!result.empty()) return result;
+    int cntp = 0;
+    struct stat st;
+    bool ok = false;
+    std::string path;
+    kinfo_file *kif = nullptr;
+    static kvm_t *kd = nullptr;
+    char errbuf[_POSIX2_LINE_MAX];
+    const char *pwd = nullptr, *penv = nullptr;
+    if (!std::string(argv0).empty()) {
+        if (argv0[0] == '/') {
+            path = argv0;
+        } else if (std::string(argv0).find('/') == std::string::npos) {
+            penv = getenv("PATH");
+            if (penv && *penv) {
+                wcstring wenv = str2wcstring(penv);
+                wcstring_list_t env = split_string(wenv, L':');
+                for (std::size_t i = 0; i < env.size(); i++) {
+                    path = wcs2string(env[i].c_str(), env[i].length()) + "/" + argv0;
+                    if (!stat(path.c_str(), &st) && (st.st_mode & S_IXUSR) && (st.st_mode & S_IFREG)) {
+                        break;
+                    }
+                }
+            }
+        } else {
+            pwd = getenv("PWD");
+            if (pwd && *pwd) {
+                path = std::string(pwd) + "/" + std::string(argv0);
+            }
+        }
+    }
+    if (path.empty()) { result = "fish"; return result; }
```

Comment:
As a general question: When does this fail/succeed?

What is the kvm_openfiles block used for? In which cases does that happen?

Your answer should probably be added as a comment.